File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/process/processdemo/cs/processdemo.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Diagnostics;
using System.Threading;

public class ProcessDemo
{
    public static void Main(String[] args) 
    {
        string appName = Environment.GetCommandLineArgs()[0];
        
        if(args.Length != 1)
        {
            Console.WriteLine("Usage: " + appName +" <executable>");
            return;
        }
        string executableFilename = args[0];
        
        Process process = new Process();
        process.StartInfo.FileName = executableFilename;
        process.Start();
        
        process.WaitForInputIdle();
        
        Thread.Sleep(1000);
        if(!process.CloseMainWindow())
        {
            process.Kill();
        }
    }
}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/process/processdemo/vb/ProcessDemo.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Diagnostics
Imports System.Threading

Namespace ProcessDemo

    Module ProcessDemo

        Public Sub Main()
            Dim args As String()
            Dim appName As String
            args = Environment.GetCommandLineArgs()
            appName = args(0)
        
            If (args.Length <> 2) Then
                Console.WriteLine("Usage: " + appName + " <executable>")
                Exit Sub
            End If

            Dim executableFilename As String
            executableFilename = args(1)

            Dim process As Process
            process = New Process
            process.StartInfo.FileName = executableFilename
            process.Start()
        
            process.WaitForInputIdle()
        
            Thread.Sleep(1000)
            If (Not process.CloseMainWindow()) Then
                process.Kill()
            End If

        End Sub

    End Module

End Namespace
